### ISSUE TITLE
IsWindowFocus(ImGuiFocusedFlags_RootAndChildWindows) will lose focus When I press the keyboard alt in  branch docking

### ISSUE THREAD

**Issue by weike20** [ocornut/imgui#4547]

**Comment by weike20** [comment]
7bfc379a  does not have this problem

**Comment by ocornut** [comment]
I cannot repro. Please provide a repro and gif and more details.

**Comment by weike20** [comment]
> I cannot repro. Please provide a repro and gif and more details.

It's my method call location error，thank you

**Comment by weike20** [comment]
> I cannot repro. Please provide a repro and gif and more details.

It's my method call location error，thank you

**Comment by weike20** [comment]
https://user-images.githubusercontent.com/15934692/133737622-c0f39532-9ba7-435a-bb23-118d872f05c3.mp4

**Comment by weike20** [comment]
>  14-31-51_Trim.mp4



> I cannot repro. Please provide a repro and gif and more details.
In the video, I was repeatedly pressing the ALT key

**Comment by PathogenDavid** [comment]
I can repro this. The window needs to be docked with another and keyboard navigation needs to be disabled.

The focus is changing to the containing dockspace as seen below:

![image](https://user-images.githubusercontent.com/278957/133826899-4115fe41-222e-466b-8485-552729e32f18.png)

The issue was introduced by a combination of 8fa502ca42c6010e26394f4ace0087c97694a760 and ff428f1d2ff852b077f1fe6eba80a11e80ccad13 (which was merged into docking in 0b06c4b2b0281a8aced61f276bb3f6ad0f6f9180.) It's specifically caused by this if statement:

https://github.com/ocornut/imgui/blob/0b06c4b2b0281a8aced61f276bb3f6ad0f6f9180/imgui.cpp#L10335-L10342

Based on `start_windowing_with_keyboard`, I believe the fix is to add a check if keyboard navigation is enabled:

```cpp
if (io.KeyMods == ImGuiKeyModFlags_Alt && (io.KeyModsPrev & ImGuiKeyModFlags_Alt) == 0 && (g.IO.ConfigFlags & ImGuiConfigFlags_NavEnableKeyboard) != 0)
```

(Previously the config flag was implicitly checked since `IsNavInputTest` uses `io.NavInputsDownDuration` which is all -1 unless `ImGuiConfigFlags_NavEnableKeyboard` is set.)

**Comment by PathogenDavid** [comment]
This issue was also somewhat exasperated by 5ee40c8d34bea3009cf462ec963225bd22067e5e since it makes it less obvious what's going on. Before that commit you'd see the nav highlight on the tab:

![image](https://user-images.githubusercontent.com/278957/133833238-15210ef0-caf5-4255-af61-d96e4bfbc8e8.png)

Also as an aside, the child window and the `ImGuiFocusedFlags_RootAndChildWindows` are red herrings. They don't matter. IE: Both of these will repro the issue:

```cpp
static void GH4547()
{
    ImGui::Begin("GH-4547");
    ImGui::BeginChild("TestChild");
    ImGui::Text("Focus: %s", ImGui::IsWindowFocused(ImGuiFocusedFlags_RootAndChildWindows) ? "Yes" : "No");
    ImGui::EndChild();
    ImGui::End();
}

static void GH4547_NoChild()
{
    ImGui::Begin("GH-4547 (No Child)");
    ImGui::Text("Focus: %s", ImGui::IsWindowFocused() ? "Yes" : "No");
    ImGui::End();
}
```

**Comment by ocornut** [comment]
Pushed fix fa0a314f59ffe317a0de9af5f292f3564ca3a566 thank you @PathogenDavid very much for the investigation!
